splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropTableConstantOperation.java

```diff
@@ -175,6 +177,12 @@ public void executeConstantAction(Activation activation) throws StandardExceptio
 
             /* Drop the store element at last, to prevent dangling reference for open cursor, beetle 4393. */
             tc.dropConglomerate(heapId);
+
+            /* is the table pinned ? , if yes we need to drop it */
+            if(td.isPined()){
+                EngineDriver.driver().processorFactory().distributedProcessor().dropPinnedTable(td.getHeapConglomerateId());
+            }
```

Comment:
Transaction semantics may break if a pin was not dropped but the table got dropped. Throw an exception to fail the transaction in this case.